dashboard/app/models/pd/payment/payment_factory.rb

```diff
@@ -0,0 +1,28 @@
+module Pd::Payment
+  module PaymentFactory
+    def self.get_payment(workshop)
+      raise 'Workshop required.' unless workshop && workshop.is_a?(Pd::Workshop)
+      get_calculator_class(workshop).try{|payment| payment.instance.calculate(workshop)}
+    end
+
+    def self.get_calculator_class(workshop)
```

Comment:
It's used in `payment_factory_test`, to verify the correct types are used without going through all the calculations every time.